### ISSUE TITLE
Emulator authentication users not found by https callable function

### ISSUE THREAD

**Issue by chrisjacobs91** [firebase/firebase-tools#3103]
### [REQUIRED] Environment info

**firebase-tools:** 9.2.2

**Platform:** macOS

### [REQUIRED] Test case

After creating a user in emulator authentication, I run a https callable function looking for that user ```admin.auth().getUser(id)``` and get the error:
> code: 'auth/user-not-found',
> message: 'There is no user record corresponding to the provided identifier.'

### [REQUIRED] Steps to reproduce
1. Start firebase emulators with the following command: ```firebase emulators:start --only firestore,auth --import=backups/data/emulator-exit-data```
2. Create authentication user in firebase emulator UI, copy the generated ID
3. Run functions with the command: ```firebase functions:shell``` (I have to do this separately as the emulators don't support calling https callable functions)
4. Trigger the https callable function with the following command in the functions shell terminal: ```main.createProfile({uid: '8DFFrqmP52TAPYCTGxl6KGny9kF5', fullName:'chris', username:'new', location:'here'})``` (the data provided isn't relevant to this error as the error happens before any of it is used).
5. Function runs with the following code (I've just copied the code up to when the error occurs). Note - I have to comment out the check for context.auth and manually define the authUID variable because it is not possible to run this function as an authenticated user from the command line (please tell me if this isn't true!)

```
export const createProfile = functions.region('europe-west1').https.onCall(async (data, context) => {
  try {
  
    // Error Handling based on request data
/*     if (!context.auth) {
      throw new functions.https.HttpsError('failed-precondition', 'The function must be called while authenticated.');
    }; */

    // Parse data
    const {uid, fullName, username, location, avatarUrl = null} = data
  
    // Get auth user info
    // const authUID:string = context.auth.uid
    const authUID:string = '8DFFrqmP52TAPYCTGxl6KGny9kF5'
    let authUserObject = null
    
    try {
      authUserObject = await admin.auth().getUser(authUID);

    } catch(e) {
      console.log(e)
      throw new functions.https.HttpsError('failed-precondition', `Authenticated user doesn't exist: ${authUID}`);
    };
```
6. Get an error:
> code: 'auth/user-not-found',
> message: 'There is no user record corresponding to the provided identifier.'

### [REQUIRED] Expected behavior
Expected behaviour is that the admin.auth().getUser() function can successfully retrieve the data for my newly created auth user.

It's worth pointing out that I have not historically had any trouble using admin.auth().getUser() in firestore triggered functions running on the emulator, so perhaps the issue is this function is running from the functions:shell rather than the emulators start command.

### [REQUIRED] Actual behavior
Error received, cannot find auth user.

**Comment by chrisjacobs91** [comment]
Oh, I commented out the admin.auth() call to debug the rest of the function while I was waiting and realised it's calling production, not the emulator.

Is there a way that I can use the https callable functions against the emulators?

**Comment by kmcnellis** [comment]
The root of this is the functions:shell command wouldn't know to set the environment variables to connect to the emulators.

As a workaround, you can follow the "Connect to the <Cloud Firestore|Authentication> emulator" steps for the Admin SDK (setting the GCLOUD_PROJECT, FIRESTORE_EMULATOR_HOST, and FIREBASE_AUTH_EMULATOR_HOST env variables) :
https://firebase.google.com/docs/emulator-suite/connect_firestore#admin_sdks and https://firebase.google.com/docs/emulator-suite/connect_auth#admin_sdks

**Comment by kmcnellis** [comment]
I think it would be a good feature request to have an emulator version of the functions:shell command, so you don't have to set those manually.

**Comment by samtstern** [comment]
Ok so just so I understand this the flow would be:

1. In one terminal you run `emulators:start` to start up service emulators like Firestore, Auth, etc.
2. In a second terminal you run `functions:shell` to interactively test Functions and you'd like those shell functions to discover the emulators running in the other process and communicate with them.

This is pretty doable, I just want to make sure I'm getting the flow right.

**Comment by samtstern** [comment]
But yes @kmcnellis has the correct short-term workaround, just set `FIREBASE_AUTH_EMULATOR_HOST="localhost:1234"` (or whatever your host is) in your shell before running `firebase functions:shell` and it will work.

**Comment by chrisjacobs91** [comment]
@samtstern pretty much, I use ```emulators:start --only firestore,auth```
otherwise when I run ```firebase functions:shell``` I get an error saying port not open, couldn't start functions emulator.

**Comment by chrisjacobs91** [comment]
Thank you for the workaround @kmcnellis! 

Now I can run my callable functions locally in this way, against the firestore and auth emulators, I've noticed two other issues:

1. Firestore and Auth triggered functions are not triggered.
2. I cannot call the callable functions with authentication. (Or perhaps I just don't know how to?) 
FYI - I throw an error if the function is called unauthenticated, which I currently have to comment out for local command line testing, and then hardcode what the context.auth.id value would have been.

**Comment by samtstern** [comment]
@chrisjacobs91 yes those are pretty core limitations of the `functions:shell` and not something we're looking to change.  In the end the shell is just an interactive sandbox and as with most sandboxes it can't simulate a whole platform.

If you want to be able to test more advanced flows I suggest running the functions together with the other emulators (in `emulators:start`) and then using one of these testing strategies:
https://github.com/firebase/quickstart-testing

Let me know if there's something you think you can't do that way.

**Comment by chrisjacobs91** [comment]
@samtstern I was going to ask, would it be easier to just create a little web page that logs a user in and can call the callable functions 😄 

thanks for all your help on this!
